lambda/recipes-responder/src/update_retrievable_processor.ts

```diff
@@ -0,0 +1,65 @@
+/* eslint @typescript-eslint/naming-convention: "off"  -- PollingAction uses a more CAPI-like convention*/
+import type {RetrievableContent} from "@guardian/content-api-models/crier/event/v1/retrievableContent";
+import {ContentType} from "@guardian/content-api-models/v1/contentType";
+import type { PollingResult} from "@recipes-api/lib/capi";
+import {callCAPI} from "@recipes-api/lib/capi";
+import {CapiKey} from "./config";
+import {handleContentUpdate} from "./update_processor";
+
+// Why are these functions in their own file? To make mocking in tests easier.
+
+// Why can't we just import this from capi module? It appears that the info gets erased at transpile time, so we end up
+//with object undefined errors at runtime :-(
+enum PollingAction {
+  CONTENT_EXISTS,
+  CONTENT_MISSING,
+  CONTENT_GONE,
+  CONCIERGE_UNHAPPY,
+  INTERNAL_BUG,
+  UNEXPECTED_RESPONSE,
+  RATE_LIMITED
+}
+
+async function retrieveContent(capiUrl:string) :Promise<PollingResult>{
+  if(!CapiKey) {
+    throw new Error("You need to set CAPI_KEY in order to make requests to CAPI");
+  }
+
+  const params = [
+    `show-fields=internalRevision`,
+    `show-blocks=all`,
+    `show-channels=all`,
+    `api-key=${CapiKey}`,
+    `format=thrift`
+  ].filter(v=>!!v).join("&");
+
+  return callCAPI(`${capiUrl}?${params}`);
+}
+
+export async function handleContentUpdateRetrievable(retrievable:RetrievableContent): Promise<number>
+{
+  if(retrievable.contentType!=ContentType.ARTICLE) return 0;  //no point processing live-blogs etc.
+
+  const capiResponse = await retrieveContent(retrievable.capiUrl);
+  switch(capiResponse.action) {
+    case PollingAction.CONTENT_EXISTS:
+      //Great, we have it - but should check if this has now been superceded
+      if(capiResponse.content?.fields?.internalRevision ?? 0 > (retrievable.internalRevision ?? 99) ) {
```

Comment:
The use of `99` here looks suspect to me as it suggests we're assuming there can never be more updates to an item than this, but if it _can_ happen, it almost certainly _will_ at some point.